File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        for(int i = 0; i < targets.length; i++) {
            
            for(int j = 0; j < targets[i].length(); j++) {
                int min = 1000;
                char c = targets[i].charAt(j);
                for(int k = 0; k < keymap.length; k++) {
                    int index = keymap[k].indexOf(targets[i].charAt(j));
                    if (index != -1) {
                        min = Math.min(min, index);
                    }
                    
                }
                if(min == 1000) {
                answer[i] = -1;
                break;
            }
            else {
                answer[i] += min + 1;
            }
                
            }
            
            
        }
        return answer;
    }
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    var answer = [];
    const map = new Map();
    for (const key of keymap) {
        for (let i = 0; i < key.length; i++) {
            const k = key[i];
            if (!map.has(k)) {
                map.set(k,i+1);
            } else {
                const min = Math.min(map.get(k),i+1);
                map.set(k,min);
            }
        }
    }
    for (const target of targets) {
        let sum = 0;
        for (const char of target) {
            if (map.get(char)) {
              sum += map.get(char);
            } else {
                answer.push(-1);
                sum = 0;
                break;
            }
        }
        if (sum > 0) answer.push(sum);
    }
    
    return answer;
}